specs/_features/reuse_indexes/beacon-chain.md

```diff
@@ -0,0 +1,65 @@
+# Reuse indexes -- The Beacon Chain
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Preset](#preset)
+  - [Time parameters](#time-parameters)
+- [Helpers](#helpers)
+  - [Predicates](#predicates)
+    - [`is_reusable_validator`](#is_reusable_validator)
+- [Beacon chain state transition function](#beacon-chain-state-transition-function)
+  - [Block processing](#block-processing)
+    - [Modified `get_index_for_new_validator`](#modified-get_index_for_new_validator)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification to assign new deposits to existing validator records that have withdrawn long ago.
+
+*Note:* This specification is built upon [Capella](../../capella/beacon_chain.md) and is under active development.
+
+## Preset
+
+### Time parameters
+
+| Name | Value | Unit | Duration |
+| - | - | - |
+| `REUSE_VALIDATOR_INDEX_DELAY` | `uint64(2**16)` (= 65,536) | epochs | ~1 year |
+
+## Helper functions
+
+### Predicates
+
+#### `is_reusable_validator`
+
+```python
+def is_reusable_validator(validator: Validator, balance: Gwei, epoch: Epoch) -> bool:
+    """
+    Check if ``validator`` index can be re-assigned to a new deposit.
+    """
+    return (
+      validator.withdrawable_epoch < epoch - REUSE_VALIDATOR_INDEX_DELAY
```

Comment:
need to handle the case where this underflows

I'd suggest just making this False when current epoch is not yet to `REUSE_VALIDATOR_INDEX_DELAY`

```python
    if epoch < REUSE_VALIDATOR_INDEX_DELAY:
        return False
    return (
        # your current conditions
    )
```